### DISCUSSION TITLE
open build locally from the filesystem in a Browser...

### DISCUSSION THREAD

**Discussion by cnichte** [thuliteio/doks#1140] [Q&A] [answered]
Hi. My idea was, to open a static website locally from the filesystem in a Browser. When I open the build page `/public/index.html`  locally in the browser, the css is missing and the clicks do not work. Seems that the browser blocks everything. Setting the Browsers Securtiy Settings to 'full disc access' changes nothing... Maybe I'm making a mistake here...  Any idea, why this is so and how to solve this? `npm run preview ` and `npm run dev` and on the web server all works fine as expected.

**Comment by h-enk** [comment, marked as answer]
It's because a **Request URL** like e.g. 
`file:///D:/main.e932f010c0b46d51657a28f8c155619c7ac59bf0678e0d67dee84f52da6dd016af85b295a2c39bbe292f0ba09a9b2cda98a8b2979a6f65584145c837d920c547.css` with **Referrer Policy** `strict-origin-when-cross-origin` (browser default) triggers a CORS error — see also: [Cross-Origin Resource Sharing (CORS)](https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS)

That's why (amongst others) Hyas (and Doks) ships with [http-server](https://github.com/http-party/http-server) (with CORS enabled) so you can [preview your site locally](https://docs.gethyas.com/guides/deploy/#previewing-your-site-locally) by running:

```bash
npm run build -- --baseURL=/ && npm run preview
```

**Reply by cnichte** [reply]
Ah, thanks that makes sense.
